src/pages/search.vue

```diff
@@ -95,14 +97,24 @@ export default defineComponent({
       handleTab,
     }
   },
+  /**
+   * asyncData blocks the rendering of the page, so we only
+   * update the state from the route here, and do not fetch media.
+   */
   async asyncData({ route, $pinia }) {
     const searchStore = useSearchStore($pinia)
-    const mediaStore = useMediaStore($pinia)
     await searchStore.initProviderFilters()
     searchStore.setSearchStateFromUrl({
       path: route.path,
       urlQuery: route.query,
     })
+  },
+  /**
+   * Fetch media, if necessary, in a non-blocking way.
+   */
+  async fetch() {
+    const mediaStore = useMediaStore()
+    const searchStore = useSearchStore()
```

Comment:
What do you think about moving this fetching to the `setup` block? I believe it also works, we could save duplicated lines of declarations and keep the code in a more standard style instead of using custom Nuxt hooks.

Reply:
As far as I understand, the `setup` block is blocking the rendering of the page, so it would mean the black loading indicator on the top of the page while loading the media items :(
I hate duplication, too. I might actually be able to return the necessary store methods from the `setup`, and use _them_ in fetch...